packages/react/src/QuestionnaireForm/QuestionnaireForm.tsx

```diff
@@ -49,14 +49,16 @@ export function QuestionnaireForm(props: QuestionnaireFormProps): JSX.Element |
   onSubmitRef.current = props.onSubmit;
 
   useEffect(() => {
-    if (questionnaireResponse?.item) {
-      setResponse((prevResponse) => updateResponseItems(prevResponse, questionnaireResponse.item, questionnaire));
+    if (questionnaireResponse?.item && questionnaire) {
+      const response = buildInitialResponse(questionnaire, questionnaireResponse);
+      const evaluatedResponse = evaluateAndMergeResponseItems(response, questionnaire);
+      setResponse(evaluatedResponse);
     } else {
       if (questionnaire && getQuestionnaireIdentity(prevQuestionnaire) === getQuestionnaireIdentity(questionnaire)) {
         return;
       }
       // throw out the existing response and start over
-      setResponse(questionnaire ? buildInitialResponse(questionnaire) : undefined);
+      setResponse(questionnaire ? buildInitialResponse(questionnaire, undefined) : undefined);
```

Comment:
Then we can clean things up like this.

```suggestion
      setResponse(questionnaire ? buildInitialResponse(questionnaire) : undefined);
```